lib/linguist/heuristics.yml

```diff
@@ -18,6 +18,7 @@
 #                       regular expression (with union).
 # and                 - An and block merges multiple rules and checks that all of
 #                       of them must match.
+# filename_pattern    - Same as pattern, but tests filenames instead of content.
```

Comment:
Oh, that. This was an addition to the heuristics format to accommodate `Text` files which ended with version numbers (stuff like `GPL-2.2` and stuff). I must've overlooked the heuristic that used it when resolving merge conflicts, but on second thought, I'm not sure it's even needed...

I recall it was @pchaigno's suggestion. @pchaigno, do we still need this? If not, we can simplify the format by removing the `filename_pattern` feature, but it might come in handy in future...